iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/distribution/SourceRewriter.java

```diff
@@ -1556,12 +1556,14 @@ private void calculateGroupByLevelNodeAttributes(
           descriptorExpressions.addAll(originalDescriptor.getOutputExpressions());
         }
 
-        for (String groupedInputExpressionString :
-            originalDescriptor.getGroupedInputExpressionStrings()) {
+        for (List<Expression> groupedInputExpressions :
+            originalDescriptor.getGroupedInputExpressions()) {
+          String groupedInputExpressionsString =
+              originalDescriptor.getInputString(groupedInputExpressions);
           List<Expression> inputExpressions =
-              childrenExpressionMap.get(groupedInputExpressionString);
+              childrenExpressionMap.get(groupedInputExpressionsString);
           if (inputExpressions != null && !inputExpressions.isEmpty()) {
-            descriptorExpressions.addAll(inputExpressions);
+            descriptorExpressions.addAll(groupedInputExpressions);
           }
         }
```

Comment:
use a field to indicate that there exists view, otherwise we keep same as previous.